snapcraft/storeapi/v2/_api_schema.py

```diff
@@ -339,3 +338,130 @@
     "required": ["channel-map", "revisions", "snap"],
     "type": "object",
 }
+
+# Version 27, found at https://dashboard.snapcraft.io/docs/v2/en/tokens.html#api-tokens-whoami
+WHOAMI_JSONSCHEMA: Dict[str, Any] = {
+    "properties": {
+        "account": {
+            "properties": {
+                "email": {
+                    "description": "The primary email for the user",
+                    "type": "string",
+                },
+                "id": {
+                    "description": "The store account ID for the user",
+                    "type": "string",
+                },
+                "name": {
+                    "description": "The display name for the user",
+                    "type": "string",
+                },
+                "username": {
+                    "description": "The store username for the user",
+                    "type": "string",
+                },
+            },
+            "required": ["email", "id", "name", "username"],
+            "type": "object",
+        },
+        "channels": {
+            "oneOf": [
+                {
+                    "description": "A list of channels to restrict the macaroon to, allows wildcards in the fnmatch format (https://docs.python.org/3/library/fnmatch.html)",
+                    "items": {"description": "Valid channel name.", "type": "string"},
+                    "minItems": 1,
+                    "type": "array",
+                    "uniqueItems": True,
+                },
+                {"type": "null"},
+            ]
+        },
+        "errors": {
+            "items": {
```

Comment:
Interesting that errors is not allowed to be null.  Is there any scenario where we should be checkin this field for errors?

Reply:
yes, we have some weird logic today for when you can login but cannot do anything until you sign the developer agreement or setup a username on the store side. This would allow for replacing that custom logic